libdebug/debugger/debugger.py

```diff
@@ -0,0 +1,407 @@
+        self: Debugger,
+        original_syscall: int | str,
+        new_syscall: int | str,
+        hook_hijack: bool = True,
+        **kwargs: int,
+    ) -> SyscallHook:
+        """Hijacks a syscall in the target process.
+
+        Args:
+            original_syscall (int | str): The syscall name or number to hijack.
+            new_syscall (int | str): The syscall name or number to replace the original syscall with.
+            hook_hijack (bool, optional): Whether the syscall after the hijack should be hooked. Defaults to True.
+            **kwargs: (int, optional): The arguments to pass to the new syscall.
+
+        Returns:
+            SyscallHook: The syscall hook object.
+        """
+        return self._internal_debugger.hijack_syscall(original_syscall, new_syscall, hook_hijack, **kwargs)
+
+    def migrate_to_gdb(self: Debugger, open_in_new_process: bool = True) -> None:
+        """Migrates the current debugging session to GDB."""
+        self._internal_debugger.migrate_to_gdb(open_in_new_process)
+
+    @property
+    def threads(self: Debugger) -> list[ThreadContext]:
+        """Get the list of threads in the process."""
+        return self._internal_debugger.threads
+
+    @property
+    def memory(self: Debugger) -> MemoryView:
+        """Get the memory view of the process."""
+        return self._internal_debugger.memory
+
+    @property
+    def breakpoints(self: Debugger) -> dict[int, Breakpoint]:
+        """Get the breakpoints set on the process."""
+        return self._internal_debugger.breakpoints
+
+    @property
+    def syscall_hooks(self: InternalDebugger) -> dict[int, SyscallHook]:
+        """Get the syscall hooks dictionary.
+
+        Returns:
+            dict[int, SyscallHook]: the syscall hooks dictionary.
+        """
+        return self._internal_debugger.syscall_hooks
+
+    @property
+    def signal_hooks(self: InternalDebugger) -> dict[int, SignalHook]:
+        """Get the signal hooks dictionary.
+
+        Returns:
+            dict[int, SignalHook]: the signal hooks dictionary.
+        """
+        return self._internal_debugger.signal_hooks
+
+    @property
+    def pprint_syscalls(self: Debugger) -> bool:
+        """Get the state of the pprint_syscalls flag.
+
+        Returns:
+            bool: True if the debugger should pretty print syscalls, False otherwise.
+        """
+        return self._internal_debugger.pprint_syscalls
+
+    @pprint_syscalls.setter
+    def pprint_syscalls(self: Debugger, value: bool) -> None:
+        """Set the state of the pprint_syscalls flag.
+
+        Args:
+            value (bool): the value to set.
+        """
+        if not isinstance(value, bool):
+            raise TypeError("pprint_syscalls must be a boolean")
+        if value:
+            self._internal_debugger.enable_pretty_print()
+        else:
+            self._internal_debugger.disable_pretty_print()
+
+        self._internal_debugger.pprint_syscalls = value
```

Comment:
The logic of enable_pretty_print and disable_pretty_print are drastically different. Hence, it makes sense to have two separated functions instead of a single setter.